src/NetscriptFunctions.ts

```diff
@@ -1836,6 +1841,12 @@ export function NetscriptFunctions(workerScript: WorkerScript): NS {
     },
     tryWritePort: function (port: any, data: any = ""): any {
       updateDynamicRam("tryWritePort", getRamCost(Player, "tryWritePort"));
+      if ((typeof data !== "string" && typeof data !== "number") || isNS(data)) {
+        throw makeRuntimeErrorMsg(
```

Comment:
I think these two lines of code (without the `isNS` bit) is all that was missing to cause the reported bug. I've heard a few people in chat say that "writePort" is fine, but "tryWritePort" is bugged.